pkg/middleware/auth_proxy.go

```diff
@@ -41,32 +53,33 @@ func initContextWithAuthProxy(store *remotecache.RemoteCache, ctx *models.ReqCon
 		return true
 	}
 
-	// Try to log in user from various providers
-	id, err := auth.Login()
+	id, err := logUserIn(auth, username, logger, false)
 	if err != nil {
-		logger.Error(
-			"Failed to login",
-			"username", username,
-			"message", err.Error(),
-			"error", err.DetailsError,
-		)
 		ctx.Handle(407, err.Error(), err.DetailsError)
 		return true
 	}
 
-	// Get full user info
+	logger.Debug("Got user ID, getting full user info", "userID", id)
+
 	user, err := auth.GetSignedUser(id)
 	if err != nil {
-		logger.Error(
-			"Failed to get signed user",
-			"username", username,
-			"message", err.Error(),
-			"error", err.DetailsError,
-		)
-		ctx.Handle(407, err.Error(), err.DetailsError)
-		return true
+		logger.Debug("Failed to get user info given ID, retrying without cache", "userID", id)
+		_ = auth.RemoveUserFromCache(logger)
```

Comment:
I think we should handle this error. It would also be nice with some comments about why we remove the cache and try again. This is quite a fragile part of the codebase so some tests would be really nice.

Reply:
@bergquist OK - do you know if all errors returned here will be hard, or if there can be some we should ignore?

Reply:
https://github.com/grafana/grafana/blob/master/pkg/infra/remotecache/remotecache.go#L18 this is the only known error I think you have to care about. 

If not sure we have to return the error. Perhaps just logging would be enough?

Reply:
@bergquist Haven't done anything about tests yet, but handling this error now, by logging it if it isn't remotecache.ErrCacheItemNotFound.